### PR TITLE
Add Nx.axis_index/2

### PR DESCRIPTION
**Pull request by jonatanklosko**
We currently use `Nx.Shape.normalize_axis/3` a lot internally, but a similar functionality could be useful outside. Specifically, when writing custom generic functions that accept `:axis`, in order to validate shapes we may need the axis index to operate on shape tuples.

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -2763,6 +2763,27 @@ defmodule Nx do
...
+
+      iex> Nx.axis_index(Nx.iota({100, 10, 20}), -1)
+      2
+
+      iex> Nx.axis_index(Nx.iota({100, 10, 20}, names: [:batch, :x, :y]), :x)
+      1
+
+  """
+  @doc type: :shape
+  def axis_index(tensor, axis) do
```
Comment (jonatanklosko) [inline nx/lib/nx.ex:2782]: Technically the returned value is also a valid axis, so I'm not sure about the naming. Alternatives: `normalized_axis`, `plain_axis`, wdyt?

**Comment by josevalim** [comment]
I think the current name is great! We just need an example of how this function raises for an invalid index.

**Comment by jonatanklosko** [comment]
@josevalim done!

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**